packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -0,0 +1,444 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { Connections, IConnectable } from './connections';
+import { CfnVPCEndpoint } from './ec2.generated';
+import { SecurityGroup } from './security-group';
+import { VpcSubnet } from './vpc';
+import { IVpcNetwork, IVpcSubnet, SubnetSelection } from './vpc-ref';
+
+/**
+ * A VPC endpoint.
+ */
+export interface IVpcEndpoint extends cdk.IConstruct {
+  /**
+   * The VPC endpoint identifier.
+   */
+  readonly vpcEndpointId: string;
+}
+
+/**
+ * A VPC gateway endpoint.
+ */
+export interface IVpcGatewayEndpoint extends IVpcEndpoint {
+  /**
+   * Exports this VPC endpoint from the stack.
+   */
+  export(): VpcGatewayEndpointImportProps;
+}
+
+/**
+ * The type of VPC endpoint.
+ */
+export enum VpcEndpointType {
+  /**
+   * Interface
+   */
+  Interface = 'Interface',
+
+  /**
+   * Gateway
+   */
+  Gateway = 'Gateway'
+}
+
+/**
+ * A VPC endpoint service.
+ */
+export interface VpcEndpointService {
```

Comment:
This needs to be an `IVpcEndpointService` (if something implements it begins with an "I").